packages/near-membrane-base/src/membrane.ts

```diff
@@ -1540,6 +1591,76 @@ export function createMembraneMarshall(
               }
             : noop;
 
+        const isForeignPointerOfObjectProto = IS_IN_SHADOW_REALM
+            ? // eslint-disable-next-line no-return-assign
+              (foreignTargetPointer: Pointer): boolean =>
+                  foreignTargetPointer ===
+                  (foreignPointerObjectProto === undefined
+                      ? (foreignPointerObjectProto = getTransferablePointer(ObjectProto))
+                      : foreignPointerObjectProto)
+            : alwaysFalse;
+
+        const isForeignPointerOfTypedArrayProto = IS_IN_SHADOW_REALM
+            ? // eslint-disable-next-line no-return-assign
```

Comment:
Lazy helpers to detect foreign pointers of objects which trigger fast object tracking to be invalidated.